src/wp-includes/class-walker-nav-menu.php

```diff
@@ -39,6 +39,30 @@ class Walker_Nav_Menu extends Walker {
 		'id'     => 'db_id',
 	);
 
+	/**
+	 * The URL to the privacy policy page.
+	 *
+	 * @since 6.2.0
+	 * @var string
+	 */
+	private static $privacy_policy_url = null;
```

Comment:
Does this need to be `static`? It seems like it could be non-static, as this would solve potential problems with unit testing down the road, as well as the unlikely scenario where the privacy policy URL changes between different nav menu instances.

Reply:
The thing is having this static allow us to share the same data across any instance of the class, if this "state" is not shared between instances we would query the database each time get_privacy_policy_url() is called. each instance of the Walker will trigger a database query

Reply:
That's exactly the thing I want to prevent with this change because the privacy url is not a thing that changes regularly or that can be different across Walker instances

Reply:
On that case I think this could be better and cleaner

```php
/**
	 * The URL to the privacy policy page.
	 *
	 * @since 6.2.0
	 * @var string
	 */
	private $privacy_policy_url = null;

	/**
	 * Constructor.
	 */
	public function __construct() {
		$this->privacy_policy_url = get_privacy_policy_url();
	}
```

Reply:
How many walker instances are there on a typical page though? Two?

Reply:
LOL I found the issue because a website I was trying to fix uses 6 `wp_nav_menu` some for mobile and some for desktop, but still I think this is needed because if a single menu has a lot of subitems only one walker instance will trigger N-menu-items queries. I'll remove the static I'm ok keeping the things simple and a little bit more performant.

Reply:
I like your suggestion in https://github.com/WordPress/wordpress-develop/pull/8140#discussion_r1920652714

Reply:
I updated the PR with the constructor approach, thanks for your feedback